lms/djangoapps/xblock_user_state/interface.py

```diff
@@ -0,0 +1,140 @@
+from xblock.fields import Scope
+
+class XBlockUserStateClient(object):
+    """
+    First stab at an interface for accessing XBlock User State. This will have
+    use StudentModule as a backing store in the default case.
+
+    Scope/Goals:
+    1. Mediate access to all student-specific state stored by XBlocks.
+        a. This includes "preferences" and "user_info" (i.e. UserScope.ONE)
+        b. This includes XBlock Asides.
+        c. This may later include user_state_summary (i.e. UserScope.ALL).
+           - Though I'm fuzzy on where we're going in general with this and how
+             it relates to content.
+        d. This may include group state in the future.
+        e. This may include other key types + UserScope.ONE (e.g. Definition)
+           - I think this implies a per-user partition scheme and not a
+             user+course partition scheme.
+    2. Assume network service semantics.
+        At some point, this will probably be calling out to an external service.
+        Even if it doesn't, we want to be able to implement circuit breakers, so
+        that a failure in StudentModule doesn't bring down the whole site.
+        This also implies that the client is running as a user, and whatever is
+        backing it is smart enough to do authorization checks.
+    3. This does not yet cover export-related functionality.
+
+    Open Questions:
+    1. Is it sufficient to just send the block_key in and extract course +
+       version info from it?
+    2. Do we want to use the username as the identifier? Privacy implications?
+       Ease of debugging?
+    3. Would a get_many_by_type() be useful?
+    """
+
+    class ServiceUnavailableError(Exception):
+        pass
+
+    class PermissionDeniedError(Exception):
+        pass
+
+    def get(self, username, block_key, scope=Scope.user_state, fields=None):
```

Comment:
Shouldn't the contracts go on the more general API here instead of the Django ORM implementation? I would hope that we could have consistent data types at this interface level.